trace/backend_collector.go

```diff
@@ -73,6 +76,39 @@ func (r witnessWithInfo) toReport(dir kgxapi.NetworkDirection) (*kgxapi.WitnessR
 	}, nil
 }
 
+func (w *witnessWithInfo) recordTimestamp(isRequest bool, t akinet.ParsedNetworkTraffic) {
+	if isRequest {
+		w.requestEnd = t.FinalPacketTime
+	} else {
+		w.responseStart = t.ObservationTime
+	}
+
+}
+
+func (w witnessWithInfo) computeProcessingLatency(isRequest bool, t akinet.ParsedNetworkTraffic) {
+	// Processing latency is the time from the last packet of the request,
+	// to the first packet of the response.
+	requestEnd := w.requestEnd
+	responseStart := t.ObservationTime
+
+	// handle arrival in opposite order
+	if isRequest {
+		requestEnd = t.FinalPacketTime
+		responseStart = w.responseStart
+	}
+
+	// Missing data, leave as default value in protobuf
+	if requestEnd.IsZero() || responseStart.IsZero() {
+		return
+	}
+
+	// HTTPMethodMetadata only for now
+	if meta := spec_util.HTTPMetaFromMethod(w.witness.Method); meta != nil {
+		latency := responseStart.Sub(requestEnd)
+		meta.ProcessingLatency = float32(latency.Microseconds()) / 1000.0
```

Comment:
How about `meta.ProcessingLatency = float32(latency/time.Millisecond)`?

Reply:
Same problem, can't divide a float by a Duration, have to convert it to a Duration first.  (And the time you get is then a Duration, which is an int64, so the fractional part would be gone.)

Reply:
`latency / time.Millisecond` does type-check, but you're right that you lose the fractional part. https://play.golang.org/p/kTyRO1PFp0b